src/routes/user/gallery/create.js

```diff
@@ -0,0 +1,42 @@
+import Joi from 'joi'
+import shortid from 'shortid'
+
+import models from '../../../models'
+
+const codexRecordSchema = Joi.object().keys({
+  _id: Joi.string().regex(/^[0-9a-f]{24}$/i).required(),
```

Comment:
Just for my learning, what's the context on renaming this every time?

Reply:
It's because I strip out the `_id` field and leave it as `id` in responses, because I think `_id` is kind of awkward. But when you pass in just `id` to a route like this, it has to be renamed to `_id` for mongoose / mongodb to store the field as an ObjectId and make the populations work later.

So renaming this here in the Joi validation allows you to just pass in an object "as previously received by the api", and prevents me from having to do something like:

```
newGalleryData.codexRecords.forEach((codexRecord) => {
  codexRecord._id = codexRecord.id
})
```

before saving the model.